pkg/aws/client/client.go

```diff
@@ -1508,6 +1508,38 @@ func (c *Client) CreateSubnet(ctx context.Context, subnet *Subnet) (*Subnet, err
 	return fromSubnet(output.Subnet), nil
 }
 
+// CreateCIDRReservation creates a EC2 subnet cidr reservation resource.
+func (c *Client) CreateCIDRReservation(ctx context.Context, subnet *Subnet, cidr string, reservationType string) (string, error) {
+	input := &ec2.CreateSubnetCidrReservationInput{
+		SubnetId:        &subnet.SubnetId,
+		Cidr:            aws.String(cidr),
+		ReservationType: aws.String(reservationType),
+	}
+
+	output, err := c.EC2.CreateSubnetCidrReservationWithContext(ctx, input)
+	if err != nil {
+		return "", err
+	}
+	return *output.SubnetCidrReservation.Cidr, nil
+}
+
+// CreateCIDRReservation gets EC2 subnet cidr reservations.
+func (c *Client) GetIPv6CIDRReservations(ctx context.Context, subnet *Subnet) ([]string, error) {
+	input := &ec2.GetSubnetCidrReservationsInput{
+		SubnetId: &subnet.SubnetId,
+	}
+
+	output, err := c.EC2.GetSubnetCidrReservationsWithContext(ctx, input)
+	if err != nil {
+		return nil, err
+	}
+	var cidrs []string
+	for _, cidrReservation := range output.SubnetIpv6CidrReservations {
+		cidrs = append(cidrs, *cidrReservation.Cidr)
```

Comment:
Actually, It would be weird to get an array containing nil values and a CIDRReservation without a Cidr would also be strange. What should I check for nil? cidrReservation, cidr or both?

Reply:
You are right, it would be pretty unusual.
But I think we're better safe than sorry, so I vote for adding a check for both `cidrReservation` and `cidrReservation.Cidr`.